src/utils/extractVendorExtensions.ts

```diff
@@ -0,0 +1,22 @@
+import { SchemaFragment } from '@stoplight/json-schema-tree';
+
+export type VendorExtensionsList = {
+  [keyof: string]: unknown;
+};
+
+export type VendorExtensionsResult = [number, VendorExtensionsList];
+
+/**
+ * Extract all vendor extensions or properties prefix with 'x-' from the schema definition
+ * @param fragment The fragment to extract the vendor extensions from
+ * @returns VendorExtensionsResult
+ */
+export function extractVendorExtensions(fragment: SchemaFragment = {}): VendorExtensionsResult {
+  const extensionKeys = Object.keys(fragment).filter(key => key.toLowerCase().startsWith('x-'));
```

Comment:
I'm not sure we should lower-case the key. If a key starts with an upper-cased X-, then IMHO it shouldn't be considered a spec ext.